### ISSUE TITLE
Raise skill cap of enchantment spells

### ISSUE THREAD

**Issue by eakaye** [NetHack/NetHack#779]
Currently, no role can raise enchantment spell skill above Skilled, and indeed only Wizards can even reach Skilled. This makes sense in current versions, where the highest enchantment spell level is 3, since high skill grants no real benefit to success chance at such low spell levels under reasonable circumstances. But 3.7.0 is rejiggering various spell levels, and in particular raising "charm monster" from level 3 to 5. This changes the math and means a player might reasonably want to raise enchantment skill over its current limits.

Wizards should certainly be able to reach Expert, to match the levels of the other "arcane" schools. Tourists also need to be able to get to Skilled, or they'd need to be level 27 (!) to reliably cast their "special" spell, the one they're supposed to be especially good at.

<sub>And let me just say that I strongly disagree with the spell level changes. They turn sleep from a useful low-level escape spell into something rather impractical, since by the time you can level and practice it to reliability, you're facing lots of undead and high-MR monsters. And most characters are ascension-ready before they're capable of casting high-level spells, so the new charm monster isn't likely to see much use even if the skill cap is raised.</sub>

**Comment by pat-rankin** [comment]
There is some method to the madness.  I'm not sure how enchantment spells skill was originally assigned, but the limits as they are now were mostly set when a bunch of features from slash'em got incorporated into 3.3.0.  Slash'EM had extra roles, including "Flame Mage", "Ice Mage", and "Necromancer" that didn't make the cut.  I suspect that the two specialized mage classes there were allowed to reach expert in enchantment spells.  Similarly, none of the roles in nethack can reach expert in pole-arms (nor is it likely that anyone would want to) but the "Yeoman" role in slash'em could.

What does this have to do with nethack as it exists today?  Not much....

**Comment by eakaye** [comment]
So the change to the Charm Monster spell level is now live, and as predicted, the spell is now effectively Wizard-only. The Tourist special spell bonus doesn't really help when they have to be XL 27 to get rid of the difficult penalty, at which point they'll certainly have a set of non-metallic armor, so the lower penalty won't do much. To make the effect useful without being overpowered for its level, I propose splitting it into two:

- The existing "charm monster" is renamed to "greater charm" (or "dire charm", or whatever), and retains its current level and effects. The scroll and magic harp continue to use this effect.
- A new spell of "charm monster" is introduced, at spell level 2. It only tames monsters strictly below level 15, so it excludes jabberwocks, minotaurs, dragons, and Titans, among others. It can _pacify_ higher-level monsters, but not tame them. So it can only potentially remove these powerful monsters as threats, not turn them into a resource (unless you level drain them first). Tourists have this as their special spell.

We already have spells of healing and extra healing, so there's at least some precedent for one spell being strictly superior to another.

**Comment by Tomsod** [comment]
Pacification is as good as taming for removing a game-ending threat, and getting a dragon pet only requires a polytrap, a wild dog, a cookie, and some patience, so your version is not that much weaker.  In fact, sometimes I _want_ a monster to be peaceful instead of tame; while the conversion is trivial, your spell would save some effort.

Although the new charm monster also allows maintaining inediate pet tameness, which is pretty neat and might justify the higher level.  This effect is also improved at Skilled, breaking the previously held implicit rule that no spells above level 4 change with skill, as being able to cast them at all is change enough.

And on the topic of skill caps in general, one of my least favorite 3.6 changes was halving the number of roles that can cast skilled versions of fireball and cone of cold, which were practically their own spells and pretty cool to use.  We now have them as invoke effects available to anyone, which is better.  Sometimes role skill limits still seem too restrictive, e.g. grappling hooks working badly below expert, available to exactly one role (I had a whole rant about that in another issue), or gnomes getting a bonus to crossbow use, despite most gnomish roles being unable to advance in crossbow skill.

**Comment by copperwater** [comment]
You're correct about the strategy implications of a spell that pacifies monsters under current game behavior, Tomsod, but I do think there is an option for a lesser-charm mechanic that doesn't have those issues: _Temporary_ pacification.

A lesser charm monster spell could make a monster peaceful, but with the effect wearing off after some time, at which point they revert to being hostile and possibly will resist being charmed by a subsequent spell.

This would obviously require a new field or two on the monst struct to track this status, but if we're adding a new spell we're not concerned about save compatibility anyway. The crudest implementation would be a single bit field "mpacified" or whatever, which like mconf/mstun/etc has some fixed chance of wearing off per turn, but it could be done with a bit more depth (and make the pacification duration a bit more reliable) by using more bits.
